### PR TITLE
Organizing the script folder

### PR DESCRIPTION
**Pull request by VahidooX**
Organized the scripts folder based on the task. This is an initial draft.

### CODE REVIEW

File: examples/asr/speech_to_label.py [hunk 1]
```diff
@@ -47,10 +47,10 @@
 # Task 2: Voice Activity Detection
 
 ## Preparing the dataset
-Use the `process_vad_data.py` script under <NEMO_ROOT>/scripts in order to prepare the dataset.
+Use the `get_speech_commands_data.py` script under <NEMO_ROOT>/scripts in order to prepare the dataset.
```
Comment (titu1994) [inline examples/asr/speech_to_label.py:50]: Vad data is not necessarily speech command data. @fayejf is this fine?

**Review by titu1994 — COMMENTED** [review]
Revert average_model_checkpoints.py. It's a general script so it should go into the base dir

Rest looks fine.

**Comment by nithinraok** [comment]
Suggestion to change the folder name from speaker_diarization to speaker_recognition in scripts.
I couldn't comment on unchanged files. Some links in speaker recognition notebooks point to these scripts. 
Some of them:

> in tutorials/Speaker_Recognition_Verification.ipynb

* get_hi-mia-data.py (https://github.com/NVIDIA/NeMo/blob/rir_noise_aug_2/scripts/get_hi-mia_data.py) script to download the necessary files -> get_hi-mia-data.py (https://github.com/NVIDIA/NeMo/blob/main/scripts/speaker_recognition/get_hi-mia_data.py) script to download the necessary files
* wget -P scripts https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/scripts/scp_to_manifest.py -> wget -P scripts https://raw.githubusercontent.com/NVIDIA/NeMo/$BRANCH/scripts/speaker_recogntion/scp_to_manifest.py

File: examples/asr/speech_to_label.py (hunk 1 above)
Comment (fayejf) [reply examples/asr/speech_to_label.py:50]: no this is not fine. _process_vad_data_ also include others light rebalance/preprare demo data for notebook, it should be _process_vad_data_ here. And I can't find the script under scripts, could you bring it back? @VahidooX

Comment (VahidooX) [reply examples/asr/speech_to_label.py]: renamed it back to process_vad_data, but it looks like that it always download this dataset: "google speech command v2", yes?
And I updated the notebooks. Would you please check that it works like this now?

**Comment by fayejf** [comment]
I've checked scripts and tutorials for speech command, VAD and speaker diarization. 

Could you please change
1.  **NeMo/scripts/** in  examples/asr/vad_infer  to **NeMo/scripts/voice_activity_detection**
2. multiple **<NEMO_ROOT>/scripts** in examples/asr/speech_to_label.py to correposinding subfolders.
3. in tutorials/asr/06 VAD: scripts/voice_activity_recognition/process_vad_data -> scripts/**dataset_processing**/process_vad_data
4. in tutorials/speaker_recognition/Speaker_Diarization_Inference.ipynb: "you can also tune the threshold on your dev set. Use this provided script" fix the link of the **script**. It would be much appreciated if you can also help change the link to https://github.com/NVIDIA/NeMo/blob/r1.0.0rc1/nemo/collections/asr/parts/vad_utils.py in the above line. 

Thanks!

File: examples/asr/speech_to_label.py (hunk 1 above)
Comment (fayejf) [reply examples/asr/speech_to_label.py]: yes. It includes downloading the google speech command v2 but also a lot of other stuff. we could further clean it up (say extract downloading part and reuse it in different tasks) but let's do it out of RC? 

Yes please see below.

**Comment by VahidooX** [comment]
> 
> 
> I've checked scripts and tutorials for speech command, VAD and speaker diarization.
> 
> Could you please change
> 
>     1. **NeMo/scripts/** in  examples/asr/vad_infer  to **NeMo/scripts/voice_activity_detection**
> 
>     2. multiple **<NEMO_ROOT>/scripts** in examples/asr/speech_to_label.py to correposinding subfolders.
> 
>     3. in tutorials/asr/06 VAD: scripts/voice_activity_recognition/process_vad_data -> scripts/**dataset_processing**/process_vad_data
> 
>     4. in tutorials/speaker_recognition/Speaker_Diarization_Inference.ipynb: "you can also tune the threshold on your dev set. Use this provided script" fix the link of the **script**. It would be much appreciated if you can also help change the link to https://github.com/NVIDIA/NeMo/blob/r1.0.0rc1/nemo/collections/asr/parts/vad_utils.py in the above line.
> 
> 
> Thanks!

Thanks for checking them out. I missed them.
1-done
2-done
3-done
4-Please check it again to make sure I put the right links. We need to change them to main after the release?

**Review by fayejf — APPROVED** [review]
LGTM. I've checked script and tutorials for speech command, VAD and speaker diarization. Could you validate the speaker recognition/verification part? Thanks!  @nithinraok

**Comment by nithinraok** [comment]
I see you are merging to rc1 branch but links are pointed to main. If we test before merging to main, notebook tutorials gonna fail

**Comment by nithinraok** [comment]
> > I see you are merging to rc1 branch but links are pointed to main. If we test before merging to main, notebook tutorials gonna fail
> 
> It looks tricky. I tried not to change those parts. The code should work with both actually as RC is merged into main everyday. So if it is based on main, it was going to get tested with main anyway and this PR is not going to affect the results. Should I change all to main or RC?

Actually I just see [one notebook](https://github.com/VahidooX/NeMo/blob/organize_scripts/tutorials/speaker_recognition/Speaker_Recognition_Verification.ipynb) script (get_hi-mia-data.py) to main branch, but rest of notebooks looks good. Also its path should be changed to `dataset_processing` not `speaker_recognition`. Just this rest all looks good to me

**Comment by VahidooX** [comment]
> 
> 
> > > I see you are merging to rc1 branch but links are pointed to main. If we test before merging to main, notebook tutorials gonna fail
> > 
> > 
> > It looks tricky. I tried not to change those parts. The code should work with both actually as RC is merged into main everyday. So if it is based on main, it was going to get tested with main anyway and this PR is not going to affect the results. Should I change all to main or RC?
> 
> Actually I just see [one notebook](https://github.com/VahidooX/NeMo/blob/organize_scripts/tutorials/speaker_recognition/Speaker_Recognition_Verification.ipynb) script (get_hi-mia-data.py) to main branch, but rest of notebooks looks good. Also its path should be changed to `dataset_processing` not `speaker_recognition`. Just this rest all looks good to me. I vote main since for later releases we don't need to change again.

Thanks for catching them! Updated those. I also prefer main, but for now, I made them all RC as we are need to change others to master later anyway.

**Review by okuchaiev — APPROVED** [review]
lgtm, thanks!

**Review by nithinraok — APPROVED** [review]
LGTM. Thanks Vahid

**Pull request merged**